From be66772bf5bd8e0cc0b4b662743e81000e3037f2 Mon Sep 17 00:00:00 2001
From: Patrick Combes <patrick.combes@intel.com>
Date: Fri, 14 Jun 2013 10:09:13 +0200
Subject: Add support for multiple bt_vendor.conf at runtime

BZ: 117478

To select the bt_vendor.conf at runtime, it is required to build the
system image with the bt_vendor.conf for every platforms we may want to
support.
Therefore the files are not more named bt_vendor.conf on the target image
but bt_baylake.conf, bt_redhookbay.conf, etc...

The right bt_"vendor".conf is informed by a property "ro.bt.conf_file"
which can be filled at boot time.

Change-Id: Ib66ac9a516a8ec44d6b261a12014cecf5d4912f0
Signed-off-by: Patrick Combes <patrick.combes@intel.com>
Reviewed-on: http://android.intel.com:8080/126529
Reviewed-by: Fert, Laurent <laurent.fert@intel.com>
Reviewed-by: Seibel, Eric <eric.seibel@intel.com>
Reviewed-by: Gros, Julien <julien.gros@intel.com>
Tested-by: Champciaux, Nicolas <nicolas.champciaux@intel.com>
Reviewed-by: cactus <cactus@intel.com>
Tested-by: cactus <cactus@intel.com>
---
 conf/intel/Android.mk              |   22 ++++++++++++++++++++++
 conf/intel/baylake/Android.mk      |   10 ++++++++++
 conf/intel/bodegabay/Android.mk    |   10 ++++++++++
 conf/intel/byt_m_crb/Android.mk    |   10 ++++++++++
 conf/intel/byt_t_ffrd10/Android.mk |   10 ++++++++++
 conf/intel/byt_t_ffrd8/Android.mk  |   10 ++++++++++
 conf/intel/ctp7160/Android.mk      |   10 ++++++++++
 conf/intel/ctpscalelt/Android.mk   |   10 ++++++++++
 conf/intel/redhookbay/Android.mk   |   10 ++++++++++
 conf/intel/saltbay_pr0/Android.mk  |   10 ++++++++++
 conf/intel/saltbay_pr1/Android.mk  |   10 ++++++++++
 conf/intel/victoriabay/Android.mk  |   10 ++++++++++
 src/bt_vendor_brcm.c               |    5 ++++-
 13 files changed, 136 insertions(+), 1 deletions(-)
 create mode 100644 conf/intel/Android.mk
 create mode 100644 conf/intel/baylake/Android.mk
 create mode 100644 conf/intel/bodegabay/Android.mk
 create mode 100644 conf/intel/byt_m_crb/Android.mk
 create mode 100644 conf/intel/byt_t_ffrd10/Android.mk
 create mode 100644 conf/intel/byt_t_ffrd8/Android.mk
 create mode 100644 conf/intel/ctp7160/Android.mk
 create mode 100644 conf/intel/ctpscalelt/Android.mk
 create mode 100644 conf/intel/redhookbay/Android.mk
 create mode 100644 conf/intel/saltbay_pr0/Android.mk
 create mode 100644 conf/intel/saltbay_pr1/Android.mk
 create mode 100644 conf/intel/victoriabay/Android.mk

diff --git a/conf/intel/Android.mk b/conf/intel/Android.mk
new file mode 100644
index 0000000..73860b4
--- /dev/null
+++ b/conf/intel/Android.mk
@@ -0,0 +1,22 @@
+LOCAL_PATH:= $(call my-dir)
+
+include $(CLEAR_VARS)
+
+LOCAL_MODULE := bt_vendor.conf
+LOCAL_MODULE_TAGS := optional
+LOCAL_REQUIRED_MODULES := \
+	bt_baylake.conf \
+	bt_byt_m_crb.conf \
+	bt_byt_t_ffrd10.conf \
+	bt_byt_t_ffrd8.conf \
+	bt_bodegabay.conf \
+	bt_ctp7160.conf \
+	bt_ctpscalelt.conf \
+	bt_redhookbay.conf \
+	bt_saltbay_pr0.conf \
+	bt_saltbay_pr1.conf \
+	bt_victoriabay.conf
+
+include $(BUILD_PHONY_PACKAGE)
+
+include $(call all-makefiles-under,$(LOCAL_PATH))
diff --git a/conf/intel/baylake/Android.mk b/conf/intel/baylake/Android.mk
new file mode 100644
index 0000000..57ceaba
--- /dev/null
+++ b/conf/intel/baylake/Android.mk
@@ -0,0 +1,10 @@
+LOCAL_PATH:= $(call my-dir)
+include $(CLEAR_VARS)
+
+LOCAL_MODULE := bt_$(notdir $(LOCAL_PATH)).conf
+$(call print-vars $(LOCAL_MODULE))
+LOCAL_MODULE_TAGS := optional
+LOCAL_MODULE_CLASS := ETC
+LOCAL_MODULE_PATH := $(TARGET_OUT_ETC)/bluetooth
+LOCAL_SRC_FILES := bt_vendor.conf
+include $(BUILD_PREBUILT)
diff --git a/conf/intel/bodegabay/Android.mk b/conf/intel/bodegabay/Android.mk
new file mode 100644
index 0000000..57ceaba
--- /dev/null
+++ b/conf/intel/bodegabay/Android.mk
@@ -0,0 +1,10 @@
+LOCAL_PATH:= $(call my-dir)
+include $(CLEAR_VARS)
+
+LOCAL_MODULE := bt_$(notdir $(LOCAL_PATH)).conf
+$(call print-vars $(LOCAL_MODULE))
+LOCAL_MODULE_TAGS := optional
+LOCAL_MODULE_CLASS := ETC
+LOCAL_MODULE_PATH := $(TARGET_OUT_ETC)/bluetooth
+LOCAL_SRC_FILES := bt_vendor.conf
+include $(BUILD_PREBUILT)
diff --git a/conf/intel/byt_m_crb/Android.mk b/conf/intel/byt_m_crb/Android.mk
new file mode 100644
index 0000000..57ceaba
--- /dev/null
+++ b/conf/intel/byt_m_crb/Android.mk
@@ -0,0 +1,10 @@
+LOCAL_PATH:= $(call my-dir)
+include $(CLEAR_VARS)
+
+LOCAL_MODULE := bt_$(notdir $(LOCAL_PATH)).conf
+$(call print-vars $(LOCAL_MODULE))
+LOCAL_MODULE_TAGS := optional
+LOCAL_MODULE_CLASS := ETC
+LOCAL_MODULE_PATH := $(TARGET_OUT_ETC)/bluetooth
+LOCAL_SRC_FILES := bt_vendor.conf
+include $(BUILD_PREBUILT)
diff --git a/conf/intel/byt_t_ffrd10/Android.mk b/conf/intel/byt_t_ffrd10/Android.mk
new file mode 100644
index 0000000..57ceaba
--- /dev/null
+++ b/conf/intel/byt_t_ffrd10/Android.mk
@@ -0,0 +1,10 @@
+LOCAL_PATH:= $(call my-dir)
+include $(CLEAR_VARS)
+
+LOCAL_MODULE := bt_$(notdir $(LOCAL_PATH)).conf
+$(call print-vars $(LOCAL_MODULE))
+LOCAL_MODULE_TAGS := optional
+LOCAL_MODULE_CLASS := ETC
+LOCAL_MODULE_PATH := $(TARGET_OUT_ETC)/bluetooth
+LOCAL_SRC_FILES := bt_vendor.conf
+include $(BUILD_PREBUILT)
diff --git a/conf/intel/byt_t_ffrd8/Android.mk b/conf/intel/byt_t_ffrd8/Android.mk
new file mode 100644
index 0000000..57ceaba
--- /dev/null
+++ b/conf/intel/byt_t_ffrd8/Android.mk
@@ -0,0 +1,10 @@
+LOCAL_PATH:= $(call my-dir)
+include $(CLEAR_VARS)
+
+LOCAL_MODULE := bt_$(notdir $(LOCAL_PATH)).conf
+$(call print-vars $(LOCAL_MODULE))
+LOCAL_MODULE_TAGS := optional
+LOCAL_MODULE_CLASS := ETC
+LOCAL_MODULE_PATH := $(TARGET_OUT_ETC)/bluetooth
+LOCAL_SRC_FILES := bt_vendor.conf
+include $(BUILD_PREBUILT)
diff --git a/conf/intel/ctp7160/Android.mk b/conf/intel/ctp7160/Android.mk
new file mode 100644
index 0000000..57ceaba
--- /dev/null
+++ b/conf/intel/ctp7160/Android.mk
@@ -0,0 +1,10 @@
+LOCAL_PATH:= $(call my-dir)
+include $(CLEAR_VARS)
+
+LOCAL_MODULE := bt_$(notdir $(LOCAL_PATH)).conf
+$(call print-vars $(LOCAL_MODULE))
+LOCAL_MODULE_TAGS := optional
+LOCAL_MODULE_CLASS := ETC
+LOCAL_MODULE_PATH := $(TARGET_OUT_ETC)/bluetooth
+LOCAL_SRC_FILES := bt_vendor.conf
+include $(BUILD_PREBUILT)
diff --git a/conf/intel/ctpscalelt/Android.mk b/conf/intel/ctpscalelt/Android.mk
new file mode 100644
index 0000000..57ceaba
--- /dev/null
+++ b/conf/intel/ctpscalelt/Android.mk
@@ -0,0 +1,10 @@
+LOCAL_PATH:= $(call my-dir)
+include $(CLEAR_VARS)
+
+LOCAL_MODULE := bt_$(notdir $(LOCAL_PATH)).conf
+$(call print-vars $(LOCAL_MODULE))
+LOCAL_MODULE_TAGS := optional
+LOCAL_MODULE_CLASS := ETC
+LOCAL_MODULE_PATH := $(TARGET_OUT_ETC)/bluetooth
+LOCAL_SRC_FILES := bt_vendor.conf
+include $(BUILD_PREBUILT)
diff --git a/conf/intel/redhookbay/Android.mk b/conf/intel/redhookbay/Android.mk
new file mode 100644
index 0000000..57ceaba
--- /dev/null
+++ b/conf/intel/redhookbay/Android.mk
@@ -0,0 +1,10 @@
+LOCAL_PATH:= $(call my-dir)
+include $(CLEAR_VARS)
+
+LOCAL_MODULE := bt_$(notdir $(LOCAL_PATH)).conf
+$(call print-vars $(LOCAL_MODULE))
+LOCAL_MODULE_TAGS := optional
+LOCAL_MODULE_CLASS := ETC
+LOCAL_MODULE_PATH := $(TARGET_OUT_ETC)/bluetooth
+LOCAL_SRC_FILES := bt_vendor.conf
+include $(BUILD_PREBUILT)
diff --git a/conf/intel/saltbay_pr0/Android.mk b/conf/intel/saltbay_pr0/Android.mk
new file mode 100644
index 0000000..57ceaba
--- /dev/null
+++ b/conf/intel/saltbay_pr0/Android.mk
@@ -0,0 +1,10 @@
+LOCAL_PATH:= $(call my-dir)
+include $(CLEAR_VARS)
+
+LOCAL_MODULE := bt_$(notdir $(LOCAL_PATH)).conf
+$(call print-vars $(LOCAL_MODULE))
+LOCAL_MODULE_TAGS := optional
+LOCAL_MODULE_CLASS := ETC
+LOCAL_MODULE_PATH := $(TARGET_OUT_ETC)/bluetooth
+LOCAL_SRC_FILES := bt_vendor.conf
+include $(BUILD_PREBUILT)
diff --git a/conf/intel/saltbay_pr1/Android.mk b/conf/intel/saltbay_pr1/Android.mk
new file mode 100644
index 0000000..57ceaba
--- /dev/null
+++ b/conf/intel/saltbay_pr1/Android.mk
@@ -0,0 +1,10 @@
+LOCAL_PATH:= $(call my-dir)
+include $(CLEAR_VARS)
+
+LOCAL_MODULE := bt_$(notdir $(LOCAL_PATH)).conf
+$(call print-vars $(LOCAL_MODULE))
+LOCAL_MODULE_TAGS := optional
+LOCAL_MODULE_CLASS := ETC
+LOCAL_MODULE_PATH := $(TARGET_OUT_ETC)/bluetooth
+LOCAL_SRC_FILES := bt_vendor.conf
+include $(BUILD_PREBUILT)
diff --git a/conf/intel/victoriabay/Android.mk b/conf/intel/victoriabay/Android.mk
new file mode 100644
index 0000000..57ceaba
--- /dev/null
+++ b/conf/intel/victoriabay/Android.mk
@@ -0,0 +1,10 @@
+LOCAL_PATH:= $(call my-dir)
+include $(CLEAR_VARS)
+
+LOCAL_MODULE := bt_$(notdir $(LOCAL_PATH)).conf
+$(call print-vars $(LOCAL_MODULE))
+LOCAL_MODULE_TAGS := optional
+LOCAL_MODULE_CLASS := ETC
+LOCAL_MODULE_PATH := $(TARGET_OUT_ETC)/bluetooth
+LOCAL_SRC_FILES := bt_vendor.conf
+include $(BUILD_PREBUILT)
diff --git a/src/bt_vendor_brcm.c b/src/bt_vendor_brcm.c
index 0476ca5..dc93b74 100644
--- a/src/bt_vendor_brcm.c
+++ b/src/bt_vendor_brcm.c
@@ -27,6 +27,7 @@
 #define LOG_TAG "bt_vendor"
 
 #include <utils/Log.h>
+#include <cutils/properties.h>
 #include "bt_vendor_brcm.h"
 #include "upio.h"
 #include "userial_vendor.h"
@@ -91,6 +92,7 @@ static const tUSERIAL_CFG userial_init_cfg =
 
 static int init(const bt_vendor_callbacks_t* p_cb, unsigned char *local_bdaddr)
 {
+    char lib_conf_file[PROPERTY_VALUE_MAX];
     ALOGI("init");
 
     if (p_cb == NULL)
@@ -115,7 +117,8 @@ static int init(const bt_vendor_callbacks_t* p_cb, unsigned char *local_bdaddr)
     userial_vendor_init();
     upio_init();
 
-    vnd_load_conf(VENDOR_LIB_CONF_FILE);
+    property_get("ro.bt.conf_file", lib_conf_file, VENDOR_LIB_CONF_FILE);
+    vnd_load_conf(lib_conf_file);
 
     /* store reference to user callbacks */
     bt_vendor_cbacks = (bt_vendor_callbacks_t *) p_cb;
-- 
1.7.4.1

